src/python/gudhi/datasets/generators/points/sphere.cc

```diff
@@ -0,0 +1,61 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hind Montassif
+ *
+ *    Copyright (C) 2021 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <gudhi/random_point_generators.h>
+#include <gudhi/Debug_utils.h>
+
+#include <CGAL/Epick_d.h>
+
+namespace py = pybind11;
+
+
+typedef CGAL::Epick_d< CGAL::Dynamic_dimension_tag > Kern;
+
+py::array_t<double> generate_points_on_sphere(size_t num_points, int dim, double radius) {
+
+    py::array_t<double> points({num_points, (size_t)dim});
+
+    py::buffer_info buf = points.request();
+    double *ptr = static_cast<double *>(buf.ptr);
+
+    GUDHI_CHECK(num_points == buf.shape[0], "Py array first dimension not matching num_points on sphere");
+    GUDHI_CHECK(dim == buf.shape[1], "Py array second dimension not matching the ambient space dimension");
+
+
+    py::gil_scoped_release release;
+    auto points_generated = Gudhi::generate_points_on_sphere_d<Kern>(num_points, dim, radius);
+
+    for (size_t i = 0; i < num_points; i++)
+        for (int j = 0; j < dim; j++)
+            ptr[i*dim+j] = points_generated[i][j];
+
+    return points;
+}
+
+PYBIND11_MODULE(sphere, m) {
+      m.attr("__license__") = "LGPL v3";
+      m.def("generate_random_points", &generate_points_on_sphere,
+          py::arg("num_points"), py::arg("dim"), py::arg("radius") = 1,
```

Comment:
sklearn calls it "n_samples" apparently. I don't know if we have other places in gudhi to compare with, to try and be consistent in the naming. (C++ doesn't count since it doesn't have named parameters)

For the radius, I used "r" in tomato to imitate scipy's cKDTree.query_ball_tree, but I like the longer "radius" here.

Reply:
It seems that we use "n_samples" in `src/python/gudhi/point_cloud/dtm.py` and `src/python/gudhi/clustering/tomato.py`, and "num_points" in `_tomato.cc`.

Reply:
`_tomato.cc` doesn't count, `num_points` is only used in the C++ part, not as a named parameter.
The one in dtm.py has a meaning that is different enough that it wouldn't matter much if we pick a different name, and tomato.py only uses the name in the doc, not in the code.
So we don't have much precedent in gudhi, I guess looking for inspiration in sklearn (n_samples apparently), numpy (I see `num` in linspace, `size` for numpy.Generator.random and `n` for multinomial but they can be tuples), scipy (`n` in some places?), pytorch, etc. I would currently favor n_samples...

Reply:
There would be the option of taking a tuple parameter `size=(n_samples,dim)` to match numpy, but that doesn't look very well suited.